Fundamentals/Code/Source-Control/index-v9.md

```diff
@@ -0,0 +1,174 @@
+
+![Typical set of Umbraco Project Folders](images/folder-structure-v9.jpg)
+
+#### The Umbraco Folder
+
+The main folder where the Umbraco CMS resides in, is the `/umbraco` one inside your project.
+
+Some of the contents change when you upgrade Umbraco, and will even be deleted and re-added on a clean and rebuild! No part of your implementation or third party packages installed should update these folders.
+
+From the gitignore above we can see that the content that will change when you upgrade Umbraco and be deleted and re-added on clean and rebuild are:
+
+* `umbraco/UmbracoWebsite`
+* `umbraco/UmbracoInstall`
+* `umbraco/UmbracoBackoffice`
+* `umbraco/config/lang`
+* `umbraco/config/appsettings-schema.json`
+
+None of these folders should be commited to github, since they will automatically be added on build. If you're working with an IDE and a Build Server, it's NuGet's and MSBuild's job to restore the correct versions of these folders for you. If you don't use an IDE and prefer the commandline you can use `dotnet restore` to manually restore the nuget packages.
+
+But these are not the only files in the umbraco folder that you should not commit to your git, some files are generated during runtime and should not be commited either, these are:
+
+* `/umbraco/data/TEMP` - This folder contains examine indexs, NuCache files and so on, these are temporary and should not be commited.
+  * `Umbraco.sdf` - If you are using SQLCE for the data store in your Umbraco site, then this file IS that datastore, it will be difficult to source control the constant changes to this file.
+* `/umbraco/Logs` - Umbraco currently uses *Serilog*, and a file will be generated in this folder containing tracelogs of your application, one JSON file for each day.
+* `/umbraco/mediacache` - *ImageSharp* ships with Umbraco and when an image is requested via the processor, e.g. to be resized or cropped, a cached version of the transformed image will be stored in this folder. (The [Imaging settings section](../../../Reference/V9-Config/ImagingSettings/index.md) allows you to determine where this cache is stored)
+
+We've now covered most of the folders within the `/umbraco` folder, however there's two left, the `/umbraco/models` folder, and the `/umbraco/PartialViewMacros`. The models folder has its own section right below, but for the `PartialViewMacros` folder, the answer to "should I commit this to git" is that it depends. If you want to change the templates within the folder, or add your own, then you should commit it to git, if you don't need to do that, you should not, the build will automatically create it and its content.
```

Comment:
```suggestion
None of these folders should be committed to github since they will automatically be added on build. If you're working with an IDE and a Build Server, it's NuGet's and MSBuild's job to restore the correct versions of these folders for you. If you don't use an IDE and prefer the command line you can use `dotnet restore` to manually restore the NuGet packages.

But these are not the only files in the Umbraco folder that you should not commit to your git, some files are generated during runtime and should not be committed either, these are:

* `/umbraco/data/TEMP` - This folder contains examine indexes, NuCache files, and so on, these are temporary and should not be committed.
  * `Umbraco.sdf` - If you are using SQLCE for the data store in your Umbraco site, then this file IS that datastore, it will be difficult to source control the constant changes to this file.
* `/umbraco/Logs` - Umbraco currently uses *Serilog*, and a file will be generated in this folder containing trace logs of your application, one JSON file for each day.
* `/umbraco/mediacache` - *ImageSharp* ships with Umbraco and when an image is requested via the processor, e.g. to be resized or cropped, a cached version of the transformed image will be stored in this folder. (The [Imaging settings section](../../../Reference/V9-Config/ImagingSettings/index.md) allows you to determine where this cache is stored)

We've now covered most of the folders within the `/umbraco` folder, however, there are two left, the `/umbraco/models` folder, and the `/umbraco/PartialViewMacros`. The model's folder has its own section right below, but for the `PartialViewMacros` folder, the answer to "should I commit this to git" is that it depends. If you want to change the templates within the folder or add your own, then you should commit it to git, if you don't need to do that, you should not, the build will automatically create it and its content.
```